### PR TITLE
ignoring currently overlapping brush stroke

### LINKED ISSUE #150: bug: brush strokes sometimes combine when not intersecting
**Describe the bug**
When drawing a brush stroke, it will combine with the previous brush stroke despite not overlapping due to how you move your hand when completing the previous brush stroke.

**To Reproduce**
1. Enable the brush tool
2. Draw a brush stroke with a forward motion, making sure your hand is intersecting the stroke when you let go of the trigger.
3. Draw a separate brush stroke far away from the first one.
4. They will be connected despite not overlapping.

**Expected behavior**
The brush stroke should be removed from the collider list when it exits, for some reason that is not happening in this case.

### PR DESCRIPTION
**Pull request by jake-white**
* Fixed a bug where very distant brush strokes could get connected, often when testing on desktop. Resolves #150

### CODE REVIEW

File: Assets/Scripts/Tools/DrawingBrush.cs [hunk 1]
```diff
@@ -21,8 +21,15 @@ public class DrawingBrush : MonoBehaviour
 		[SerializeField]
 		private GameObject substrokePrefab;
 
+		private OverlapTracker overlapTracker;
+
 		public bool IsDrawing { get; set; }
 
+		private void Awake()
+		{
+			overlapTracker = GetComponent<OverlapTracker>();
```
Comment (arocull) [inline Assets/Scripts/Tools/DrawingBrush.cs:31]: Can we make this a required component so if someone modifies the tools they don't accidentally bork it: https://docs.unity3d.com/6000.0/Documentation/ScriptReference/RequireComponent.html

**Review by arocull — COMMENTED** [review]
potential edge-case we should test: if someone draws Line A, then tries to add a stroke B to it but they stop drawing while their hand is intersecting it, does this prevent the stroke from connecting stroke B to stroke A? Would this be better solved with physics layers?

File: Assets/Scripts/Tools/DrawingBrush.cs (hunk 1 above)
Comment (jake-white) [reply Assets/Scripts/Tools/DrawingBrush.cs:31]: It was already required and on the tool palette, but added

**Comment by jake-white** [comment]
> potential edge-case we should test: if someone draws Line A, then tries to add a stroke B to it but they stop drawing while their hand is intersecting it, does this prevent the stroke from connecting stroke B to stroke A? Would this be better solved with physics layers?

This edge case doesn't happen currently. Layers might work better but for now I think this is a needed quick fix. As part of #152 I'm reworking brush containers to fix the parenting/combining behavior and may make this more elegant / easy to implement

**Review by arocull — APPROVED** [review]
Tested and it works fine, thank you!
